### ISSUE TITLE
Vulkan backend device selection does not select my nvidia graphics card.

### ISSUE THREAD

**Issue by Leonz5288** [taichi-dev/taichi#4175]
**Describe the bug**
When running mpm88, Vulkan's performance is as low as CPU's. Therefore, we checked nvidia-smi and discovered that this script does not really runs on GPU. After removing the first device in `devices` in `VulkanDeviceCreator::pick_physical_device`, Vulkan would be able to run on GPU.

**To Reproduce**
Run mpm88 with high computation load in Vulkan backend. 

```py
# sample code here
```

**Log/Screenshots**
Please post the **full log** of the program (instead of just a few lines around the error message, unless the log is > 1000 lines). This will help us diagnose what's happening. For example:

```
$ python my_sample_code.py
[Taichi] mode=release
[Taichi] version 0.6.29, llvm 10.0.0, commit b63f6663, linux, python 3.8.3
...
```

**Additional comments**
If possible, please also consider attaching the output of command `ti diagnose`. This produces the detailed environment information and hopefully helps us diagnose faster.

If you have local commits (e.g. compile fixes before you reproduce the bug), please make sure you first make a PR to fix the build errors and then report the bug.

**Comment by bobcao3** [comment]
Can you run the tool from https://vulkan.gpuinfo.org/ and list all the devices found on your system?

**Comment by Leonz5288** [comment]
[GPU0] NVIDIA GeForce RTX 3090
[GPU1] llvmpipe (LLVM 12.0.0, 256 bits)

Sorry for the late reply...

**Comment by bobcao3** [comment]
This should be fixed around v0.8
